### PR TITLE
docs: refactored directory and added config.json

### PR DESCRIPTION
**Pull request by iamayushdas**
Added 

1. config.json file in order to create sidebar

2. also I have refactored the docs directory in docusaurus style 

cc @spacewander @juzhiyuan

### CODE REVIEW

File: docs/en/latest/the-internal-of-apisix-java-plugin-runner.md [hunk 1]
```diff
@@ -43,16 +66,16 @@ You can refer to the [ext-plugin.fbs](https://github.com/api7/ext-plugin-proto/b
 apisix-java-plugin-runner and APISIX use a private binary protocol for coding and decoding.
 The protocol format is
 
-```
+```shell
```
Comment (juzhiyuan) [inline docs/en/latest/the-internal-of-apisix-java-plugin-runner.md:69]: No, this is just a Text IMO

File: docs/en/latest/the-internal-of-apisix-java-plugin-runner.md [hunk 2]
```diff
@@ -43,16 +66,16 @@ You can refer to the [ext-plugin.fbs](https://github.com/api7/ext-plugin-proto/b
...
 ```
 
 The type can be 0 ~ 7, and the length can be [0, 8M). The length of data is determined by length.
 
 The current type takes the following values
 
-* 0 means error
-* 1 means prepare_conf
-* 2 means http_req_call
+- 0 means error
```
Comment (juzhiyuan) [inline docs/en/latest/the-internal-of-apisix-java-plugin-runner.md:77]: Your IDE format this?

File: docs/en/latest/how-it-works.md [hunk 3]
```diff
@@ -37,7 +60,7 @@ and APISIX communicate between processes and must be consistent.
 
 Note: If you see some error logs like
 
-```
+```shell
```
Comment (juzhiyuan) [inline docs/en/latest/how-it-works.md:63]: not sure if it's a shell

File: docs/en/latest/the-internal-of-apisix-java-plugin-runner.md [hunk 4]
```diff
@@ -43,16 +66,16 @@ You can refer to the [ext-plugin.fbs](https://github.com/api7/ext-plugin-proto/b
...
+- 1 means prepare_conf
+- 2 means http_req_call
```
Comment (juzhiyuan) [inline docs/en/latest/the-internal-of-apisix-java-plugin-runner.md:79]: Please use `*` as other files :)

**Review by juzhiyuan — COMMENTED** [review]
Others LGTM

File: docs/en/latest/the-internal-of-apisix-java-plugin-runner.md (hunk 4 above)
Comment (iamayushdas) [reply docs/en/latest/the-internal-of-apisix-java-plugin-runner.md]: check now

**Comment by spacewander** [comment]
@dmsolr 
How to exclude config.json from the license check?

**Review by juzhiyuan — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**